src/routes/movies.py

```
@@ -1,13 +1,163 @@
+    await db.commit()
+    await db.refresh(new_movie)
+
+    full_movie = await db.execute(
+        select(MovieModel)
+        .options(
+            joinedload(MovieModel.country),
+            joinedload(MovieModel.genres),
+            joinedload(MovieModel.actors),
+            joinedload(MovieModel.languages),
+        )
+        .where(MovieModel.id == new_movie.id)
+    )
+    full_movie_obj = full_movie.unique().scalar_one()
+    return MovieReadSchema.model_validate(full_movie_obj)
+
+
+@router.get("/movies/{movie_id}/", response_model=MovieReadSchema)
+async def read_movie(
+    movie_id: int, db: AsyncSession = Depends(get_db)
+):
+    movie = await get_movie(db, movie_id)
+    if not movie:
+        raise HTTPException(
+            status_code=404, detail="Movie with the given ID was not found."
+        )
+    return movie
+
+
+@router.delete("/movies/{movie_id}/", status_code=204)
+async def remove_film(
+    movie_id: int, db: AsyncSession = Depends(get_db)
+):
+    deleted_movie = await delete_movie(db, movie_id)
+    if not deleted_movie:
+        raise HTTPException(
+            status_code=404, detail="Movie with the given ID was not found."
+        )
+    return
+
+
+@router.patch("/movies/{movie_id}/")
+async def update_movie(
+    movie_id: int = Path(..., ge=1),
+    update_data: MovieUpdateSchema = Body(),
+    db: AsyncSession = Depends(get_db),
+):
+    result = await db.execute(
+        select(MovieModel).where(MovieModel.id == movie_id)
+    )
+    db_movie = result.scalar_one_or_none()
+
+    if not db_movie:
+        raise HTTPException(
+            status_code=404, detail="Movie with the given ID was not found."
+        )
+
+    update_dict = update_data.model_dump(exclude_unset=True)
+
+    try:
+        for field, value in update_dict.items():
+            setattr(db_movie, field, value)
+        await db.commit()
+    except Exception:
+        raise HTTPException(status_code=400, detail="Invalid input data.")
```

The route wraps the commit in a broad try/except that catches all exceptions and returns 400. That is okay as a safety net, but it will also swallow programming/database errors and convert them to the generic "Invalid input data." message. Prefer to validate input before setting attributes (using Pydantic validators) and only use the exception handler to catch expected validation/constraint errors. This makes debugging easier if unexpected exceptions occur during DB operations.